intertidal/mosaics.py

```diff
@@ -0,0 +1,193 @@
+    "The function will add on {product}/{version}/continental_mosaics/{year}--P1Y to the provided ouput_dir ",
+)
+@click.option(
+    "--dataset_maturity",
+    type=str,
+    default="final",
+    help="Dataset maturity metadata to used in the name of the input datasets and output dataset. "
+    "Defaults to 'final', can also be 'interim'.",
+)
+def make_mosaic_cli(
+    product,
+    band,
+    year,
+    version,
+    product_dir,
+    output_dir,
+    dataset_maturity="final",
+    sunglint="",
+):
+    input_params = locals()
+    run_id = f"[{version}] [{year}] [{band}]"
+    log = configure_logging(run_id)
+
+    # Record params in logs
+    log.info(f"{run_id}: Using parameters {input_params}")
+
+    s3 = _is_s3(product_dir)
+    product_dir = product_dir.replace("s3://", "")
+    product_dir = product_dir.rstrip("/")
+    product_dir = f"{product_dir}/{product}/{version}"
+
+    log.info(f"{run_id}: Using product_dir {product_dir}")
+    output_dir = output_dir.rstrip("/")
+    output_dir = f"{output_dir}/{product}/{version}/continental_mosaics/{year}--P1Y"
+
+    run_id = f"[{product}] [{band}] [{year}]"
+    log = configure_logging(run_id)
+
+    fs = s3fs.S3FileSystem(anon=True)
+    # Configure S3
+    configure_s3_access(cloud_defaults=True, aws_unsigned="--aws_unsigned")
+    print(
+        f"{product_dir}/**/{year}--P1Y/{product}_*{year}--P1Y_{dataset_maturity}_{band}.tif"
+    )
+
+    if s3:
+        cogs = fs.glob(
+            f"{product_dir}/**/**/{year}--P1Y/{product}_*{year}--P1Y_{dataset_maturity}_{band}.tif"
+        )
+    else:
+        cogs = glob.glob(
+            f"{product_dir}/**/{year}--P1Y/{product}_*{year}--P1Y_{dataset_maturity}_{band}.tif",
+            recursive=True,
+        )
+
+    if len(sunglint) > 0:
+        sunglint = sunglint + "_"
+    log.info(f"{run_id}: number of cogs to mosaic {len(cogs)}")
+    if len(cogs) > 0:
+        with tempfile.TemporaryDirectory() as temp_dir:
+            temp_location = Path(temp_dir)
+            log.info(f"{run_id}: writing to {temp_location}")
+            file_list_name = os.path.join(
+                temp_location, f"{product}_{year}_{band}_{version}.txt"
+            )
+            vrt_name = os.path.join(
+                temp_location, f"{product}_{year}_{band}_{version}.vrt"
+            )
+            output_name = os.path.join(
+                temp_location, f"{product}_{year}_{band}_{sunglint}{version}.tif"
+            )
+            output_file_path = os.path.join(
+                output_dir, f"{product}_{year}_{band}_{sunglint}{version}.tif"
+            )
+            log.info(f"{run_id}: Generating file {output_name}")
+            log.info(f"{run_id}: output_file_path {output_file_path}")
+
+            with open(file_list_name, "w") as f:
+                for cog in cogs:
+                    cog = cog.replace(
+                        "dea-public-data-dev/",
```

Comment:
Likely don't want to hard-code this

Reply:
We can look at doing this later.
I did also put in another hard coded replace, to enable it to work on s3 prod